handlers/showHandlers.js

```diff
@@ -0,0 +1,146 @@
+import Show from '../models/show';
+import { stringSearch } from '../helpers/stringSearch';
+
+const getShowById = async (req, res) => {
+  const { id } = req.params;
+  const show = await Show.find({ show_id: +id });
+
+  if (show.length > 0) {
+    res.json({
+      statusCode: 200,
+      message: 'Movie fetched successfully',
+      data: show
+    });
+  } else {
+    res.status(404).json({
+      statusCode: 404,
+      error: `No movie found with id ${id}`,
+      params: req.params
+    });
+  }
+};
+
+const getAllShows = (req, res) => {
+  const {
+    page,
+    director,
+    title,
+    actor,
+    country,
+    year,
+    rating,
+    duration,
+    category,
+    description,
+    type
+  } = req.query;
+  const query = {};
+
+  if (title) {
+    query.title = stringSearch(title);
+  }
+
+  if (director) {
+    query.director = stringSearch(director);
+  }
+
+  if (actor) {
+    query.cast = stringSearch(actor);
+  }
+
+  if (country) {
+    query.country = stringSearch(country);
+  }
+
+  if (year) {
+    query.release_year = +year;
+  }
+
+  if (rating) {
+    query.rating = stringSearch(rating);
+  }
+
+  if (duration) {
+    query.duration = stringSearch(duration);
+  }
+
+  if (category) {
+    query.listed_in = stringSearch(category);
+  }
+
+  if (description) {
+    query.description = stringSearch(description);
+  }
+
+  if (type) {
+    query.type = stringSearch(type);
+  }
+
+  if (page) {
+    let ITEMS_PER_PAGE = 20;
+    let startIndex = page * ITEMS_PER_PAGE - ITEMS_PER_PAGE; // Page 1 starts at index 0
+    let totalPages;
+    let remainingPages;
+
+    // Query to get total number of objects for the query to calculate totalPages and remainingPages
+    Show.where(query).countDocuments((err, count) => {
+      totalPages = Math.ceil(count / ITEMS_PER_PAGE);
+      remainingPages = totalPages - page;
+
+      Show.find(query)
+        .skip(startIndex)
+        .limit(ITEMS_PER_PAGE)
+        .then(shows => {
+          if (shows.length > 0) {
+            const moviePhrase = shows.length > 1 ? `Movies` : `Movie`;
```

Comment:
Hmmm... I've seen this somewhere else.  What can we do about that?

Reply:
Just break it out into a separate function and return either "Movie" or "Movies" depending on the input value? 

I find it easier to simply some parts of my code afterwards. Is that normal? I guess this is where refactoring comes into play.